hummingbot/market/kucoin/kucoin_api_user_stream_data_source.py

```diff
@@ -56,89 +54,58 @@ async def get_listen_key(self):
                 data: Dict[str, str] = await response.json()
                 return data
 
-    async def ping_listen_key(self, ws: websockets.WebSocketClientProtocol) -> bool:
+    async def _subscribe_topic(self, ws: websockets.WebSocketClientProtocol) -> AsyncIterable[str]:
         try:
-            pong_waiter = await ws.ping()
-            await asyncio.wait_for(pong_waiter, timeout=self.PING_TIMEOUT)
-        except asyncio.TimeoutError:
-            self.logger().warning(f"Failed to recieve pong response from server.")
-            return False
-        return True
-
-    async def _inner_messages(self, ws: websockets.WebSocketClientProtocol) -> AsyncIterable[str]:
-        try:
-            while True:
-                msg = await ws.recv()
-                self._last_recv_time = time.time()
-                yield msg
+            subscribe_request = {
+                "id": int(time.time()),
+                "type": "subscribe",
+                "topic": "/spotMarket/tradeOrders",
+                "privateChannel": True,
+                "response": True}
+            await ws.send(ujson.dumps(subscribe_request))
         except asyncio.CancelledError:
             raise
         except Exception:
-            self.logger().warning("Message recv() failed. Going to reconnect...", exc_info=True)
             return
 
-    async def messages(self) -> AsyncIterable[str]:
-        try:
-            async with (await self.get_ws_connection()) as ws:
-                async for msg in self._inner_messages(ws):
-                    yield msg
-        except asyncio.CancelledError:
-            raise
-
     async def get_ws_connection(self) -> websockets.WebSocketClientProtocol:
         stream_url: str = f"{self._current_endpoint}?token={self._current_listen_key}&acceptUserMessage=true"
-        self.logger().info(f"Reconnecting to {stream_url}.")
+        self.logger().info(f"Connecting to {stream_url}.")
 
         # Create the WS connection.
-        return websockets.connect(stream_url)
+        return websockets.connect(stream_url, ping_interval=40, ping_timeout=self.PING_TIMEOUT)
+
+    async def _inner_messages(self, ws: websockets.WebSocketClientProtocol) -> AsyncIterable[str]:
+        while True:
+            try:
+                msg: str = await ws.recv()
+                self._last_recv_time = time.time()
+                yield msg
+            except ConnectionClosed:
```

Comment:
You don't need to except these 2 exceptions if you don't do anything with them apart from raising them again.

Reply:
True. I'll remove them now.

Reply:
Done.